File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3SiteAssetDuration/QuoteContractOption3SiteAssetDurationSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace QuoteContractOption3SiteAssetDurations
    {
        public class QuoteContractOption3SiteAssetDurationSQL
        {
            private Sys.Database _database;

            public QuoteContractOption3SiteAssetDurationSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView QuoteContractOption3SiteAssetDuration_GetAll_ForQuoteContractSiteID(int QuoteContractSiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID);
                var dt = _database.ExecuteSP_DataTable("QuoteContractOption3SiteAssetDuration_GetAll_ForSiteID");
                dt.TableName = Sys.Enums.TableNames.tblQuoteContractOption3SiteAsset.ToString();
                return new DataView(dt);
            }

            public void QuoteContractOption3SiteAssetDuration_Delete(int QuoteContractSiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID);
                _database.ExecuteSP_NO_Return("QuoteContractOption3SiteAssetDuration_Delete");
            }

            public QuoteContractOption3SiteAssetDuration Insert(QuoteContractOption3SiteAssetDuration oQuoteContractOption3SiteAsset)
            {
                _database.ClearParameter();
                AddQuoteContractOption3SiteAssetParametersToCommand(ref oQuoteContractOption3SiteAsset);
                oQuoteContractOption3SiteAsset.SetQuoteContractSiteAssetDurationID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOption3SiteAssetDuration_Insert"));
                oQuoteContractOption3SiteAsset.Exists = true;
                return oQuoteContractOption3SiteAsset;
            }

            private void AddQuoteContractOption3SiteAssetParametersToCommand(ref QuoteContractOption3SiteAssetDuration oQuoteContractOption3SiteAsset)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@QuoteContractSiteID", oQuoteContractOption3SiteAsset.QuoteContractSiteID, true);
                    withBlock.AddParameter("@AssetID", oQuoteContractOption3SiteAsset.AssetID, true);
                    withBlock.AddParameter("@ScheduledMonth", oQuoteContractOption3SiteAsset.ScheduledMonth, true);
                    withBlock.AddParameter("@VisitDuration", oQuoteContractOption3SiteAsset.VisitDuration, true);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/Quotes/QuoteContractsOption3/QuoteContractOption3SiteAssetDuration/QuoteContractOption3SiteAssetDurationSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace QuoteContractOption3SiteAssetDurations

        Public Class QuoteContractOption3SiteAssetDurationSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function QuoteContractOption3SiteAssetDuration_GetAll_ForQuoteContractSiteID(ByVal QuoteContractSiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("QuoteContractOption3SiteAssetDuration_GetAll_ForSiteID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblQuoteContractOption3SiteAsset.ToString
                Return New DataView(dt)
            End Function

            Public Sub QuoteContractOption3SiteAssetDuration_Delete(ByVal QuoteContractSiteID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@QuoteContractSiteID", QuoteContractSiteID)
                _database.ExecuteSP_NO_Return("QuoteContractOption3SiteAssetDuration_Delete")
            End Sub

            Public Function [Insert](ByVal oQuoteContractOption3SiteAsset As QuoteContractOption3SiteAssetDuration) As QuoteContractOption3SiteAssetDuration
                _database.ClearParameter()
                AddQuoteContractOption3SiteAssetParametersToCommand(oQuoteContractOption3SiteAsset)

                oQuoteContractOption3SiteAsset.SetQuoteContractSiteAssetDurationID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("QuoteContractOption3SiteAssetDuration_Insert"))
                oQuoteContractOption3SiteAsset.Exists = True
                Return oQuoteContractOption3SiteAsset
            End Function

            Private Sub AddQuoteContractOption3SiteAssetParametersToCommand(ByRef oQuoteContractOption3SiteAsset As QuoteContractOption3SiteAssetDuration)
                With _database
                    .AddParameter("@QuoteContractSiteID", oQuoteContractOption3SiteAsset.QuoteContractSiteID, True)
                    .AddParameter("@AssetID", oQuoteContractOption3SiteAsset.AssetID, True)
                    .AddParameter("@ScheduledMonth", oQuoteContractOption3SiteAsset.ScheduledMonth, True)
                    .AddParameter("@VisitDuration", oQuoteContractOption3SiteAsset.VisitDuration, True)

                End With
            End Sub

#End Region

        End Class

    End Namespace

End Namespace


